app/dao/failed_logins_dao.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime, timedelta
+
+from sqlalchemy import desc
+
+from app import db
+from app.models import FailedLogin
+
+
+def dao_get_failed_logins():
+    return FailedLogin.query.all()
```

Comment:
This could end up returning hundreds of thousands of rows, looking below it only seems to be used by the tests, I would suggest removing this functionality in favour of only requesting a specific IP and/or time range

Reply:
I've removed it and instead of using this functionality for the tests I now use the dao_get_count_of_all_failed_logins_for_ip function to return the count of records in the table for the assertions.